clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1153,7 +1155,9 @@ private void handleListOffsetResponse(ListOffsetsResponse listOffsetsResponse,
         for (CompletedFetch completedFetch : completedFetches) {
             exclude.add(completedFetch.partition);
         }
-        return subscriptions.fetchablePartitions(tp -> !exclude.contains(tp));
+        // Need to clear the buffer first otherwise it will contain results from the previous fetch
+        fetchablePartitionsResultBuffer.clear();
```

Comment:
I love this approach; however, could you kindly add some justification comments on why `fetchablePartitionsResultBuffer` should be a shared state here?

Reply:
Added the comment requested. Please let me know if further changes are required.